### PR TITLE
Announcements upgrade base

### LINKED ISSUE #419: Announcements Upgrade
## Describe your idea for a new feature
* Re-develop the way announcements are created to include a **title** and **content** within the administrator panel.
* On the IRIDA users page, have an icon on the menu with a badge for the number of unread announcements.  Clicking on the icon would reveal a dropdown listing the unread announcements by title.  Clicking on the title would open the full announcement in a modal window (this would mark the announcement as read).
* Administrators could mark an announcement as a "priority" announcement, which would immediately open the announcement into a modal window when the user logs in. This modal should not be dismissable by keyboard. (this feature moved to #913)

## Describe how you would access this feature?
* The creating a new announcement should be accomplished through a newly developed admin console (see #105)
* Listing announcements should within the navigation bar on the users page.

## Additional information
Add any other context, files, or screenshots about the feature request here.

### PR DESCRIPTION
**Pull request by ksierks**
## Description of changes
- added title and priority fields to announcements
- created a script to populate the new title field from announcement content
- updated the admin announcement pages with new fields
- unread announcements rewritten with React on user dashboard
- displayed the title on user read announcements

## Related issue
Fixes #419

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [x] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
* [x] Tests added (or description of how to test) for any new features.
* [x] User documentation updated for UI or technical changes.

### CODE REVIEW

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/announcements/AnnouncementsController.java [hunk 1]
```diff
@@ -129,19 +127,27 @@ public String getControlCentreAdminPage(final Model model) {
...
+     * @param priority
+     *                  The priority of the updated announcement
      * @param model
      *                  The model for the view
      * @return A redirect to the announcement control center page
      */
     @RequestMapping(value = "/{announcementID}/details", method = RequestMethod.POST)
     @PreAuthorize("hasRole('ROLE_ADMIN')")
     public String submitUpdatedAnnouncement(@PathVariable long announcementID,
+                                            @RequestParam(defaultValue = "title placeholder") String title,
```
Comment (tom114) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/announcements/AnnouncementsController.java:143]: Is this default required here anymore?

File: src/main/resources/scripts/announcements/generate_json.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+# populated title and modified message content.
+#
+# Step #1: Run this script to generate a json file with the modified
+# announcement data.
+#
+# Step #2: Please review the json file to make sure the title was parsed
+# correctly from the message content.
+#
+# Step #3: To update the announcement table with the json file run the
+# update_announcement.py script next.
```
Comment (tom114) [inline src/main/resources/scripts/announcements/generate_json.py:18]: Please add a note in UPGRADING.md about this script.  You should consider adding something to the Upgrade Notes docs page about it if the notice in UPGRADING.md gets too long.  https://phac-nml.github.io/irida-documentation/administrator/upgrades/

**Review by tom114 — CHANGES_REQUESTED** [review]
Thanks @ksierks !  This looks great!  One question about a default parameter below an a request to add something to the upgrading guide.  Otherwise nearly good to go!

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/announcements/AnnouncementsController.java (hunk 1 above)
Comment (ksierks) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/announcements/AnnouncementsController.java]: Right, I forgot about that placeholder. I removed the deprecated constructor and placeholder. Thanks! Please see ec4627d3f9a48108c7d1ab27900b6a3a0f6cfca3.

File: src/main/resources/scripts/announcements/generate_json.py (hunk 2 above)
Comment (ksierks) [reply src/main/resources/scripts/announcements/generate_json.py:18]: I hope this works. Please see cd235201d1c5181778d4c564fe81992004722137.

**Review by tom114 — APPROVED** [review]

**Pull request merged**